### PR TITLE
fix: show shallow permissions followed by nested in alphabetical order

### LINKED ISSUE #1011: Permission ordering should show shallow permissions first
### Preliminary Checks

- [X] I have reviewed the documentation: [https://unkey.dev/docs](https://unkey.dev/docs)
- [X] I have searched for existing issues: [https://github.com/unkeyed/unkey/issues](https://github.com/unkeyed/unkey/issues)
- [X] This issue is not a question, general help request, or anything other than a bug report directly related to Unkey. Please ask questions in our Discord community: [https://unkey.dev/discord](https://unkey.dev/discord).

### Reproduction / Replay Link (Optional)

*No response*

### Issue Summary

All non-nested permissions should show before nested permissions

CleanShot 2024-02-20 at 16 47 57@2x

### Steps to Reproduce

1. (for example) Went to ...
2. Clicked on...
3. ...

### Expected behavior

The order should be:

* domain
  * delete_domain
  * read_domain
  * update_domain
  * dns
    * create_record
    * delete_record
    * read_record
    * update_record

### Other information

*No response*

### Screenshots

*No response*

### Version info

```markdown
- OS:
- Node:
- npm:
```

### PR DESCRIPTION
**Pull request by DeepaPrasanna**
## What does this PR do?

Fixes #1011
Before: 

Screenshot 2024-10-08 154904

After: 

Screenshot 2024-10-08 155136

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- Create some shallow permissions and nested ones. Now, shallow permissions should be ordered first than nested in alphabetical order.

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [Contributing Guide](./CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand areas
- [ ] Ran `pnpm build`
- [x] Ran `pnpm fmt`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

## Summary by CodeRabbit

- **New Features**
	- Introduced a sorting mechanism for permissions in the RolesPage component, enhancing the organization and readability of permission listings.

- **Bug Fixes**
	- Improved the handling of permission names by sorting them based on length and alphabetical order, ensuring a more intuitive user experience.

### CODE REVIEW

File: apps/dashboard/app/(app)/authorization/roles/[roleId]/page.tsx [hunk 1]
```diff
@@ -43,8 +43,19 @@ export default async function RolesPage(props: Props) {
...
+    if (aParts.length !== bParts.length) {
+      return aParts.length - bParts.length;
+    }
+
+    return a.name.localeCompare(b.name);
+  });
+  
   const nested: NestedPermissions = {};
-  for (const permission of workspace.permissions) {
+  for (const permission of sortedPermissions) {
```
[the remark this answers is not in the document]
Comment (chronark) [reply apps/dashboard/app/(app)/authorization/roles/[roleId]/page.tsx:58]: not needed for now

**Review by chronark — APPROVED** [review]

**Review by chronark — COMMENTED** [review]
In your "after" screenshot, shouldn't  the nested `a` be above `api`?


CleanShot 2024-10-08 at 15 46 15@2x

I think we want single permissions above nested, but then sort alphabetically

**Comment by DeepaPrasanna** [comment]
Found out that the long permission text overflows
https://www.loom.com/share/52309e9420f84310835321ee376fc128?sid=e312da88-b8a8-40b8-a023-621cac53101f

**Comment by DeepaPrasanna** [comment]
Is this good?
image

**Review by chronark — APPROVED** [review]

**Comment by DeepaPrasanna** [comment]
Hello @chronark, it seems even if you enabled auto merge few days ago, this wasn't merged. Is there anything else I could do with respect to this issue?

**Pull request merged**